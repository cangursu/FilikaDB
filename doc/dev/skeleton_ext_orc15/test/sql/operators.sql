-- comparison
SELECT '120'::ext_orc15 > '3c'::ext_orc15;
SELECT '120'::ext_orc15 >= '3c'::ext_orc15;
SELECT '120'::ext_orc15 < '3c'::ext_orc15;
SELECT '120'::ext_orc15 <= '3c'::ext_orc15;
SELECT '120'::ext_orc15 <> '3c'::ext_orc15;
SELECT '120'::ext_orc15 = '3c'::ext_orc15;

-- comparison equals
SELECT '120'::ext_orc15 > '120'::ext_orc15;
SELECT '120'::ext_orc15 >= '120'::ext_orc15;
SELECT '120'::ext_orc15 < '120'::ext_orc15;
SELECT '120'::ext_orc15 <= '120'::ext_orc15;
SELECT '120'::ext_orc15 <> '120'::ext_orc15;
SELECT '120'::ext_orc15 = '120'::ext_orc15;

-- comparison negation
SELECT NOT '120'::ext_orc15 > '120'::ext_orc15;
SELECT NOT '120'::ext_orc15 >= '120'::ext_orc15;
SELECT NOT '120'::ext_orc15 < '120'::ext_orc15;
SELECT NOT '120'::ext_orc15 <= '120'::ext_orc15;
SELECT NOT '120'::ext_orc15 <> '120'::ext_orc15;
SELECT NOT '120'::ext_orc15 = '120'::ext_orc15;

--commutator and negator
BEGIN;
CREATE TABLE ext_orc15_test AS
SELECT i::ext_orc15 as val FROM generate_series(1,10000) i;
CREATE INDEX ON ext_orc15_test(val);
ANALYZE;
SET enable_seqscan TO off;
EXPLAIN (COSTS OFF) SELECT * FROM ext_orc15_test where NOT val < 'c1';
EXPLAIN (COSTS OFF) SELECT * FROM ext_orc15_test where NOT 'c1' > val;
EXPLAIN (COSTS OFF) SELECT * FROM ext_orc15_test where 'c1' > val;
-- hash aggregate
SET enable_seqscan TO on;
EXPLAIN (COSTS OFF) SELECT val, COUNT(*) FROM ext_orc15_test GROUP BY 1;
ROLLBACK;