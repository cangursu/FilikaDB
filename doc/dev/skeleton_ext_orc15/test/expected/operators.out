-- comparison
SELECT '120'::ext_orc15 > '3c'::ext_orc15;
 ?column? 
----------
 t
(1 row)

SELECT '120'::ext_orc15 >= '3c'::ext_orc15;
 ?column? 
----------
 t
(1 row)

SELECT '120'::ext_orc15 < '3c'::ext_orc15;
 ?column? 
----------
 f
(1 row)

SELECT '120'::ext_orc15 <= '3c'::ext_orc15;
 ?column? 
----------
 f
(1 row)

SELECT '120'::ext_orc15 <> '3c'::ext_orc15;
 ?column? 
----------
 t
(1 row)

SELECT '120'::ext_orc15 = '3c'::ext_orc15;
 ?column? 
----------
 f
(1 row)

-- comparison equals
SELECT '120'::ext_orc15 > '120'::ext_orc15;
 ?column? 
----------
 f
(1 row)

SELECT '120'::ext_orc15 >= '120'::ext_orc15;
 ?column? 
----------
 t
(1 row)

SELECT '120'::ext_orc15 < '120'::ext_orc15;
 ?column? 
----------
 f
(1 row)

SELECT '120'::ext_orc15 <= '120'::ext_orc15;
 ?column? 
----------
 t
(1 row)

SELECT '120'::ext_orc15 <> '120'::ext_orc15;
 ?column? 
----------
 f
(1 row)

SELECT '120'::ext_orc15 = '120'::ext_orc15;
 ?column? 
----------
 t
(1 row)

-- comparison negation
SELECT NOT '120'::ext_orc15 > '120'::ext_orc15;
 ?column? 
----------
 t
(1 row)

SELECT NOT '120'::ext_orc15 >= '120'::ext_orc15;
 ?column? 
----------
 f
(1 row)

SELECT NOT '120'::ext_orc15 < '120'::ext_orc15;
 ?column? 
----------
 t
(1 row)

SELECT NOT '120'::ext_orc15 <= '120'::ext_orc15;
 ?column? 
----------
 f
(1 row)

SELECT NOT '120'::ext_orc15 <> '120'::ext_orc15;
 ?column? 
----------
 t
(1 row)

SELECT NOT '120'::ext_orc15 = '120'::ext_orc15;
 ?column? 
----------
 f
(1 row)

--commutator and negator
BEGIN;
CREATE TABLE ext_orc15_test AS
SELECT i::ext_orc15 as val FROM generate_series(1,10000) i;
CREATE INDEX ON ext_orc15_test(val);
ANALYZE;
SET enable_seqscan TO off;
EXPLAIN (COSTS OFF) SELECT * FROM ext_orc15_test where NOT val < 'c1';
                           QUERY PLAN                           
----------------------------------------------------------------
 Index Only Scan using ext_orc15_test_val_idx on ext_orc15_test
   Index Cond: (val >= 'c1'::ext_orc15)
(2 rows)

EXPLAIN (COSTS OFF) SELECT * FROM ext_orc15_test where NOT 'c1' > val;
                           QUERY PLAN                           
----------------------------------------------------------------
 Index Only Scan using ext_orc15_test_val_idx on ext_orc15_test
   Index Cond: (val >= 'c1'::ext_orc15)
(2 rows)

EXPLAIN (COSTS OFF) SELECT * FROM ext_orc15_test where 'c1' > val;
                           QUERY PLAN                           
----------------------------------------------------------------
 Index Only Scan using ext_orc15_test_val_idx on ext_orc15_test
   Index Cond: (val < 'c1'::ext_orc15)
(2 rows)

-- hash aggregate
SET enable_seqscan TO on;
EXPLAIN (COSTS OFF) SELECT val, COUNT(*) FROM ext_orc15_test GROUP BY 1;
            QUERY PLAN            
----------------------------------
 HashAggregate
   Group Key: val
   ->  Seq Scan on ext_orc15_test
(3 rows)

ROLLBACK;
